compiler/rustc_trait_selection/src/traits/project.rs

```diff
@@ -1282,6 +1285,7 @@ fn assemble_candidates_from_trait_def<'cx, 'tcx>(
     let bounds = match *obligation.predicate.self_ty().kind() {
         ty::Projection(ref data) => tcx.bound_item_bounds(data.item_def_id).subst(tcx, data.substs),
         ty::Opaque(def_id, substs) => tcx.bound_item_bounds(def_id).subst(tcx, substs),
+        ty::TyAlias(def_id, substs) => tcx.bound_item_bounds(def_id).subst(tcx, substs),
```

Comment:
This would probably be a breaking change (as it would use bounds on the `type` alias) but I doubt it works, I suppose it should be handled by the `ty::Infer(ty::TyVar(_))` case.

My logic being that I'm pretty sure you can write `type GetAssoc<T> = <T as Trait>::Assoc;` and that really needs to be handled like `<T as Trait>::Assoc`, not something else.